src/charm.py

```diff
@@ -238,7 +249,7 @@ def get_env_vars(self) -> dict:
             env_vars["LP_PATCH_STORAGE_POSTGRES_CONNECTION_STRING"] = postgres_patch_storage_dsn
 
         # remove empty environment values
-        env_vars = {key: value for key, value in env_vars.items() if value}
+        env_vars = {key: value for key, value in env_vars.items() if (not isinstance(value, str) or value)}
```

Comment:
Would `if value != ''` do the job more easily?

Reply:
Nice catch. Done.